transaction/src/main/java/org/apache/geode_examples/transaction/Example.java

```diff
@@ -0,0 +1,146 @@
+
+  final int id;
+
+  Example(int id) {
+    this.id = id;
+  }
+
+  static String constructJVMPath() {
+    StringBuilder builder = new StringBuilder();
+    builder.append(System.getProperty("java.home"));
+    builder.append(File.separator);
+    builder.append("bin");
+    builder.append(File.separator);
+    builder.append("java");
+    if (System.getProperty("os.name").toLowerCase().contains("win")) {
+      builder.append("w.exe");
+    }
+    return builder.toString();
+  }
+
+  static Process executeChildProcess(int id) throws IOException {
+    String[] command = new String[5];
+    command[0] = constructJVMPath();
+    command[1] = "-classpath";
+    command[2] = System.getProperty("java.class.path");
+    command[3] = "org.apache.geode_examples.transaction.Example";
+    command[4] = Integer.toString(id);
+    return Runtime.getRuntime().exec(command);
+  }
+
+  public static void main(String[] args) {
+    // connect to the locator using default port 10334
+    ClientCache cache = new ClientCacheFactory().addPoolLocator("127.0.0.1", 10334)
+        .set("log-level", "WARN").create();
+
+    // create a local region that matches the server region
+    ClientRegionFactory<String, Integer> clientRegionFactory =
+        cache.createClientRegionFactory(ClientRegionShortcut.PROXY);
+    Region<String, Integer> region = clientRegionFactory.create(REGION_NAME);
+
+    // If this is the parent process...
+    if (0 == args.length) {
+      // Initialize the region.
+      region.put(KEY, 0);
+
+      System.out.println("Expected value of counter: " + (NUMBER_OF_INCREMENTERS * INCREMENTS));
+
+      List<Process> processes = new ArrayList<>();
+
+      for (int i = 0; i < NUMBER_OF_INCREMENTERS; ++i) {
+        try {
+          processes.add(executeChildProcess(i + 1));
+        } catch (IOException ioe) {
+          ioe.printStackTrace();
+        }
+      }
+
+      for (Process process : processes) {
+        try {
+          process.waitFor();
+        } catch (InterruptedException ie) {
+          ie.printStackTrace();
+        }
+      }
+
+      System.out.println("Actual value of counter:   " + region.get(KEY));
+
+      cache.close();
+
+      return;
+    }
+
+    // If this is a child process...
+    Example example = new Example(Integer.parseInt(args[0]));
+    example.incrementEntry(region);
+
+    cache.close();
+  }
+
+  void incrementEntry(Region<String, Integer> region) {
+    ClientCache cache = (ClientCache) region.getRegionService();
+    CacheTransactionManager cacheTransactionManager = cache.getCacheTransactionManager();
+    int retries = 0;
+    for (int i = 0; i < INCREMENTS; ++i) {
+      try {
+
+        boolean incremented = false;
+        while (!incremented) {
+          try {
+            cacheTransactionManager.begin();
+            final Integer oldValue = region.get(KEY);
+            final Integer newValue = oldValue + 1;
+            if (region.replace(KEY, oldValue, newValue)) {
```

Comment:
This is weird that you are only committing if replace succeeds? If it fails you loop and call begin again without rolling back or committing? I'm not sure if the behavior of begin is defined in that case.

Using region.replace and a transaction here is basically wearing a belt and suspenders. Either one would get you the consistency you want here. Since you are demonstrating transactions, maybe just use a put?